difforig config/auto/opengl.pm lib/Parrot/Configure/Step/Methods.pm

revised to add -L/usr/X11R6/lib and /usr/bin/glut32.dll directly

diff -u config/auto/opengl.pm config/auto/opengl.pm
--- config/auto/opengl.pm	2008-06-02 17:35:05.000000000 +0000
+++ config/auto/opengl.pm	2008-07-19 08:18:54.609375000 +0000
@@ -109,11 +109,16 @@
 
  XXXX: No details yet
 
+=head3 Cygwin/w32api
 
-=head3 cygwin
+F<opengl>, F<w32api>
 
- XXXX: No details yet
+=head3 Cygwin/X
+
+F<freeglut>, F<libglut-devel>, F<xorg-x11-devel>
 
+This is detected if freeglut is installed and DISPLAY is set.
+It requires an X server.
 
 =cut
 
@@ -163,6 +168,11 @@
         conf            => $conf,
         osname          => $osname,
         cc              => $cc,
+        # Prefer Cygwin/w32api over Cygwin/X, but use X when DISPLAY is set
+        ($^O eq 'cygwin') ?
+         ($ENV{DISPLAY} ? (cygwin => '-lglut -L/usr/X11R6/lib -lGLU -lGL')
+                        : (cygwin => '/usr/bin/glut32.dll -lglu32 -lopengl32')) 
+         : (),
         win32_gcc       => '-lglut32 -lglu32 -lopengl32',
         win32_nongcc    => 'opengl32.lib glu32.lib glut32.lib',
         darwin          => '-framework OpenGL -framework GLUT',
diff -u lib/Parrot/Configure/Step/Methods.pm.orig lib/Parrot/Configure/Step/Methods.pm
--- lib/Parrot/Configure/Step/Methods.pm.orig	2008-06-24 19:59:15.000000000 +0000
+++ lib/Parrot/Configure/Step/Methods.pm	2008-07-16 19:01:24.078125000 +0000
@@ -119,6 +119,7 @@
         cc              => $cc,
         win32_gcc       => '-lalpha32 -lalpha32 -lopenalpha32',
         win32_nongcc    => 'alpha.lib',
+        cygwin          => '-lalpha32 -lXalpha32', # optional
         darwin          => 'alphadarwin.lib',
         default         => '-lalpha',
     } );
@@ -138,6 +139,8 @@
 
 =item * MSWin32 with any C-compiler other than F<gcc>.
 
+=item * Cygwin to override Mingw.
+
 =item * Darwin.
 
 =back
@@ -173,7 +176,7 @@
 
 =item * C<win32_gcc>
 
-Libraries to be added where OS is mswin32 and C-compiler is F<gcc>.
+Libraries to be added where OS is mswin32 or cygwin and C-compiler is F<gcc>.
 Single whitespace-delimited string.
 
 =item * C<win32_nongcc>
@@ -181,6 +184,12 @@
 Libraries to be added where OS is mswin32 and C-compiler is not F<gcc>.
 Single whitespace-delimited string.
 
+=item * C<cygwin>
+
+Optional libraries to be added where OS is cygwin. This overrides C<win32_gcc>
+if defined.
+Single whitespace-delimited string.
+
 =item * C<darwin>
 
 Libraries to be added where OS is Darwin.  Do not supply a value if the value
@@ -197,7 +206,8 @@
     my $args = shift;
     croak "_add_to_libs() takes hashref: $!" unless ref($args) eq 'HASH';
     my $platform =
-          (($args->{osname} =~ /mswin32/i ||
+           $args->{osname} =~ /cygwin/i      ? 'cygwin'
+        :(($args->{osname} =~ /mswin32/i ||
            $args->{osname} =~ /cygwin/i) &&
            $args->{cc} =~ /^gcc/i)          ? 'win32_gcc'
         :  $args->{osname} =~ /mswin32/i    ? 'win32_nongcc'
